controllers/teleport_master_controller/teleport_master_controller.cpp

```diff
@@ -0,0 +1,144 @@
+/*
+ * This file is part of the NUbots Codebase.
+ *
+ * The NUbots Codebase is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The NUbots Codebase is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2021 NUbots <nubots@nubots.net>
+ */
+ 
+// Special thanks to Hamburg Bit-Bots for ideas for this controller
+// https://github.com/bit-bots/wolfgang_robot/blob/feature/recognition/wolfgang_webots_sim/src/wolfgang_webots_sim/webots_camera_controller.py
+
+#include <array>
+#include <fstream>
+#include <iostream>
+#include <math.h>
+#include <random>
+#include <webots/Supervisor.hpp>
+
+#include "yaml-cpp/yaml.h"
+
+// AxisAngle rotations will be read from a config file and saved here
+std::vector<std::array<double, 4>> rotations;
+double minDistance;
+double zHeight;
+double xSize;
+double ySize;
```

Comment:
Is there a reason you made these globals?

Reply:
Not really, I can pull them inside main

Reply:
Destroy all global variables! 
![image](https://user-images.githubusercontent.com/26209642/118575342-97eaf780-b7c9-11eb-8d56-1212320a0834.png)

Reply:
Globals are bad and should be avoided at almost all costs. I'm not sure where you should be storing this state but in general globals are just asking for trouble

Reply:
Since this controller only consists of a single main function there is no reason not to just put them in main, and no benefit gained by making them global

Reply:
There's never a benefit to be gained by making variables global! Only future pain :'(

Reply:
The immediate convenience is certainly a gain 😛